daft/datatype.py

```diff
@@ -32,6 +33,43 @@
         _RAY_DATA_EXTENSIONS_AVAILABLE = False
 
 
+class ImageMode(str, Enum):
+    """
+    Supported image modes for Daft's image type.
+    """
+
+    B = "1"
+    L = "L"
+    P = "P"
+    LA = "LA"
+    RGB = "RGB"
+    RGBA = "RGBA"
+    CMYK = "CMYK"
+    YCbCr = "YCbCr"
+    LAB = "LAB"
+    HSV = "HSV"
+    I = "I"
+    F = "F"
+    L16 = "L16"
+    LA16 = "LA16"
+    RGB16 = "RGB16"
+    RGBA16 = "RGBA16"
+    RGB32F = "RGB32F"
+    RGBA32 = "RGBA32F"
+
+    @classmethod
+    def from_mode_string(cls, mode: str) -> ImageMode:
+        if mode == "1":
```

Comment:
Compatibility with typical PIL constructions for 1-bit black-and-white image mode.